SpatialGDK/Source/SpatialGDK/Private/Utils/ComponentReader.cpp

```diff
@@ -119,38 +120,36 @@ void ComponentReader::ApplySchemaObject(Schema_Object* ComponentObject, UObject*
 			if (Cmd.Type == ERepLayoutCmdType::DynamicArray)
 			{
 				UArrayProperty* ArrayProperty = Cast<UArrayProperty>(Cmd.Property);
-				bool bProcessedArray = false;
 
-				// Check if this is a FastArraySerializer array so we can simulate the FFastArraySerializerItem PreReplicatedRemove and PostReplicatedAdd calls.
-				if (UStructProperty* ParentStruct = Cast<UStructProperty>(Parent.Property))
+				// Check if this is a FastArraySerializer array and if so, call our custom delta serialization
+				if (IsFastArraySerializeProperty(ArrayProperty, Parent.Property))
 				{
-					if (ParentStruct->Struct->IsChildOf(FFastArraySerializer::StaticStruct()))
+					TArray<uint8> ValueData = GetBytesFromSchema(ComponentObject, FieldId);
```

Comment:
This looks to be duplicated a bit from the SpatialReceiver.cpp. Could it be cleaned up in a nice way?